client/src/sim.rs

```diff
@@ -138,6 +146,14 @@ impl Sim {
         self.place_block_pressed = true;
     }
 
+    pub fn select_material(&mut self, idx: usize) {
+        self.selected_material = if idx < self.material_palette.len() {
+            self.material_palette[idx]
+        } else {
+            self.material_palette[0]
+        };
```

Comment:
This could be more concise:
```suggestion
        self.selected_material = self.material_palette.get(idx).unwrap_or(self.material_palette[0]);
```